src/cached_bytes.rs

```diff
@@ -0,0 +1,129 @@
+use std::time::{Duration, Instant};
+use std::fmt;
+use std::error;
+
+use bytes::Bytes;
+use reqwest::StatusCode;
+
+type CacheUpdateFn = fn (&str) -> Result<Bytes, Box<dyn error::Error>>;
+
+#[derive(Debug)]
+pub enum MartaError {
+    Unauthorized,
+    InternalServerError,
+    GenericError(StatusCode)
+}
+
+impl fmt::Display for MartaError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match self {
+            MartaError::Unauthorized => write!(f, "{}", "Authorization failed against MARTA API"),
+            MartaError::InternalServerError => write!(f, "{}", "MARTA API did not return results"),
+            MartaError::GenericError(code) => write!(f, "MARTA API returned HTTP {}", code)
+        }
+    }
+}
+
+impl error::Error for MartaError {}
+
+pub struct CachedBytes {
+    expires_at: Instant,
+    ttl:  Duration,
+    url:  String,
+    data: Bytes,
+    updater: CacheUpdateFn
+}
+
+
+impl CachedBytes {
+    pub fn new<S: Into<String>>(url: S, ttl: Duration) -> CachedBytes {
+        CachedBytes::new_with_updater(url, ttl, |endpoint| {
```

Comment:
new default update function got a little more complicated; it now does error checking against what we get back from the URL.